File: frontend/src/types/parse.ts
export interface ParsedTransaction {
  amount: string | null;
  date: string | null;
  description: string | null;
  type: string | null;
  category_hint: string | null;
  confidence: "high" | "medium" | "low";
}

export interface BulkParseResponse {
  transactions: ParsedTransaction[];
  count: number;
}

export type PasteResult =
  | { kind: "single"; data: ParsedTransaction }
  | { kind: "bulk"; data: BulkParseResponse };

File: api/app/schemas/parse.py
# api/app/schemas/parse.py
from decimal import Decimal
from typing import Literal
from pydantic import BaseModel


class ParsedTransaction(BaseModel):
    amount: Decimal | None = None
    date: str | None = None          # ISO: YYYY-MM-DD
    description: str | None = None
    type: str | None = None          # income | expense | transfer
    category_hint: str | None = None
    confidence: Literal["high", "medium", "low"]


class BulkParseResponse(BaseModel):
    transactions: list[ParsedTransaction]
    count: int
